implementations/rust/ockam/ockam_node/src/router/stop_worker.rs

```diff
@@ -0,0 +1,53 @@
+use super::{AddressRecord, NodeState, Router};
+use crate::tokio::sync::mpsc::Sender;
+use crate::{
+    error::Error,
+    relay::{RelayMessage, PROC_ADDR_SUFFIX},
+    NodeReply, NodeReplyResult,
+};
+
+use ockam_core::{Address, Result};
+
+pub(super) async fn exec(
+    router: &mut Router,
+    addr: &Address,
+    reply: &Sender<NodeReplyResult>,
+) -> Result<()> {
+    trace!("Stopping worker '{}'", addr);
```

Comment:
There is control channel for worker too, is it used?

Reply:
It is used, but only for aborts (i.e. if a worker hasn't shut down yet by the end of the node shutdown timeout).